python/tank/authentication/sso_saml2/core/sso_saml2_core.py

```diff
@@ -222,54 +315,97 @@ def acceptNavigationRequest(self, frame, request, n_type):  # noqa
-        # the night theme of the SG Desktop impacting it. White is the safest
-        # background color.
-        self._view.setStyleSheet("background-color:white;")
+            self._view = QtWebKit.QWebView(self._dialog)
+            self._view.setPage(TKWebPageQt4(self._dialog, self._developer_mode))
+            self._view.page().networkAccessManager().authenticationRequired.connect(
+                self.on_authentication_required
+            )
+        else:
+            self._profile = QtWebEngineWidgets.QWebEngineProfile.defaultProfile()
+            self._logger.debug(
+                "Initial WebEngineProfile storage location: %s",
+                self._profile.persistentStoragePath(),
+            )
+            self._view = QtWebEngineWidgets.QWebEngineView(self._dialog)
+            self._view.setPage(
+                TKWebPageQt5(self._profile, self._dialog, self._developer_mode)
+            )
+            self._view.page().authenticationRequired.connect(
+                self.on_authentication_required
+            )
 
-        # The context : in some special cases, Shotgun will take you into an alternate
-        # login flow. E.g. when you need to change your password, enter a 2FA value,
-        # link your SSO account with an existing account on the site, etc.
-        #
-        # The issue : when using a non-approved browser, Shotgun will display a
-        # warning stating that your browser is not supported. This is fine should
-        # you be interacting with the whole site. But in our case, we only
-        # navigate the login flow; presenting relatively simple pages. The warning
-        # is not warranted. The browser used is dependent on the version of Qt/PySide
-        # being used and we have little or no control over it.
-        #
-        # The solution : hide the warning by overriding the CSS of the page.
-        # Fixing Shotgun to recognize the user-agent used by the different version
-        # of Qt so that the warning is not displayed would be a tedious task. The
-        # present solution is simpler, with the only drawback being the dependency
-        # on the name of the div for the warning. No error is generated
-        # if that div.browser_not_approved is not present in the page.
-        #
-        # Worst case scenario : should Shotgun modify how the warning is displayed
-        # it would show up in the page.
-        css_style = base64.b64encode(
-            "div.browser_not_approved { display: none !important; }"
-        )
-        url = QtCore.QUrl("data:text/css;charset=utf-8;base64," + css_style)
-        self._view.settings().setUserStyleSheetUrl(url)
+        self._view.urlChanged.connect(self._on_url_changed)
+        self._layout.addWidget(self._view)
+        self._dialog.resize(800, 600)
+
+        if QtWebKit:
+            self._logger.debug("We are in a Qt4 environment, Getting the cookie jar.")
```

Comment:
I totally agree with you.

Following our discussion, we have agreed that this could go into a later PR. I have personally committed myself to doing that work, but at a later date.